eng/scripts/Create-ApiReview.ps1

```diff
@@ -0,0 +1,185 @@
+                    Write-Host "Package version is not GA. Ignoring API view approval status"
+                }
+                elseif (!$pkgInfo.ReleaseStatus -or $pkgInfo.ReleaseStatus -eq "Unreleased")
+                {
+                    Write-Host "Release date is not set for current version in change log file for package. Ignoring API review approval status since package is not yet ready for release."
+                }
+                else
+                {
+                    # Return error code if status code is 201 for new data plane package
+                    # Temporarily enable API review for spring SDK types. Ideally this should be done be using 'IsReviewRequired' method in language side
+                    # to override default check of SDK type client
+                    if (($pkgInfo.SdkType -eq "client" -or $pkgInfo.SdkType -eq "spring") -and $pkgInfo.IsNewSdk)
+                    {
+                        if ($respCode -eq '201')
+                        {
+                            Write-Host "Package version $($version) is GA and automatic API Review is not yet approved for package $($PackageName)."
+                            Write-Host "Build and release is not allowed for GA package without API review approval."
+                            Write-Host "You will need to queue another build to proceed further after API review is approved"
+                            Write-Host "You can check http://aka.ms/azsdk/engsys/apireview/faq for more details on API Approval."
+                        }
+                        else 
+                        {
+                            Write-Host "Failed to create API Review for package $($PackageName). Please reach out to Azure SDK engineering systems on teams channel and share this build details."
+                        }
+                        return 1
+                    }
+                    else {
+                        Write-Host "API review is not approved for package $($PackageName), however it is not required for this package type so it can still be released without API review approval."
+                    }
+                }
+            }
+            else {
+                Write-Host "Build is triggered from $($SourceBranch) with prerelease version. Skipping API review status check."
+            }
+        }
+    }
+    else {
+        Write-Host "No package is found in artifact path to submit review request"
+    }
+    return 0
+}
+
+. (Join-Path $PSScriptRoot ".." common scripts common.ps1)
+
+$responses = @{}
+foreach ($artifact in $ArtifactList)
+{
+    Write-Host "Processing $($artifact.name)"
+    $result = ProcessPackage -PackageName $artifact.name
+    $responses[$artifact.name] = $result 
+}
+
+foreach($pkg in $responses.keys)
+{
+    $exitCode = 0
+    if ($responses[$pkg] -eq 1)
+    {
+        Write-Host "API changes are not approved for $($pkg)"
+        $exitCode = 1
+    }
+    exit $exitCode
```

Comment:
This exit should be moved to the end of the file so we can get a full list of all the not approved apis.

Reply:
Thanks for this catch. What a stupid I am. My intention was to exit outside but coded other way.  hmmmm